core/src/main/java/fr/sncf/osrd/api/StandaloneSimulationEndpoint.java

```diff
@@ -303,14 +313,28 @@ static SimulationResultTrain from(
             var routeOccupancies = new HashMap<String, SimulationResultRouteOccupancy>();
             double lastPosition = 0;
             for (var routePath : rjsTrainPath.routePath) {
-                var routeId = routePath.route.id.id;
+                var route = routePath.route.getRoute(infra.routeGraph.routeMap);
+                var conflictedRoutes = route.getConflictedRoutes();
                 var newPosition = lastPosition;
                 for (var trackRange : routePath.trackSections)
                     newPosition += Math.abs(trackRange.end - trackRange.begin);
                 var routeOccupancy =
                         SimulationResultRouteOccupancy.from(lastPosition, newPosition, envelope, trainLength);
-                assert !routeOccupancies.containsKey(routeId);
-                routeOccupancies.put(routeId, routeOccupancy);
+
+                for (var conflictedRoute : conflictedRoutes)
+                    if (routeOccupancies.containsKey(conflictedRoute.id)) {
+                        var oldOccupancy = routeOccupancies.get(conflictedRoute.id);
+                        assert Double.compare(oldOccupancy.timeHeadFree, routeOccupancy.timeHeadOccupy) == 0;
+                        assert Double.compare(oldOccupancy.timeTailFree, routeOccupancy.timeTailOccupy) == 0;
+                        var newOccupancy = new SimulationResultRouteOccupancy(
+                                oldOccupancy.timeHeadOccupy,
+                                routeOccupancy.timeHeadFree,
+                                oldOccupancy.timeTailOccupy,
+                                routeOccupancy.timeTailFree);
+                        routeOccupancies.replace(conflictedRoute.id, newOccupancy);
+                    } else {
+                        routeOccupancies.put(conflictedRoute.id, routeOccupancy);
+                    }
```

Comment:
Can you add  braces around the for loop. You can revert the condition and use `continue` to remove 1 level of indentation.